bundles/org.openhab.core/src/main/java/org/openhab/core/i18n/I18nException.java

```diff
@@ -0,0 +1,171 @@
+     * @param cause the cause (which is saved for later retrieval by the getCause() method). A null value is permitted,
+     *            and indicates that the cause is nonexistent or unknown.
+     * @param msgParams the optional arguments of the message defined in the properties file
+     */
+    public I18nException(String message, @Nullable Throwable cause, @Nullable Object @Nullable... msgParams) {
+        super(I18nUtil.isConstant(message) ? null : message, cause);
+        if (I18nUtil.isConstant(message)) {
+            this.msgKey = I18nUtil.stripConstant(message);
+            this.msgParams = msgParams;
+        } else {
+            this.msgKey = "";
+        }
+    }
+
+    /**
+     *
+     * @param cause the cause (which is saved for later retrieval by the getCause() method).
+     */
+    public I18nException(Throwable cause) {
+        super(cause);
+        this.msgKey = "";
+    }
+
+    /**
+     * Setup the internationalization support
+     *
+     * In case the message starts with "@text/", one of both setupI18n() methods needs to be called before using the
+     * getMessage() and getLocalizedMessage() methods.
+     *
+     * The getLocalizedMessage() method will then consider the English message.
+     *
+     * @param bundle the bundle containing the i18n properties
+     * @param i18nProvider the translation provider
+     */
+    public void setupI18n(Bundle bundle, TranslationProvider i18nProvider) {
+        setupI18n(bundle, i18nProvider, Locale.ENGLISH);
+    }
+
+    /**
+     * Setup the internationalization support
+     *
+     * In case the message starts with "@text/", one of both setupI18n() methods needs to be called before using the
+     * getMessage() and getLocalizedMessage() methods.
+     *
+     * The getLocalizedMessage() method will then consider the language provided by the locale parameter.
+     *
+     * @param bundle the bundle containing the i18n properties
+     * @param i18nProvider the translation provider
+     * @param locale the language to use for localizing the message
+     */
+    public void setupI18n(Bundle bundle, TranslationProvider i18nProvider, Locale locale) {
+        this.bundle = bundle;
+        this.i18nProvider = i18nProvider;
+        this.locale = locale;
+    }
+
+    @Override
+    public @Nullable String getMessage() {
+        Bundle localeBundle = this.bundle;
+        TranslationProvider localI18nProvider = this.i18nProvider;
+        if (msgKey.isBlank()) {
+            return super.getMessage();
+        } else if (localeBundle == null || localI18nProvider == null) {
+            logger.warn("Internationalization support is not setup for the exception");
+            return super.getMessage();
+        } else {
+            return localI18nProvider.getText(localeBundle, msgKey, null, Locale.ENGLISH, msgParams);
+        }
+    }
+
+    @Override
+    public @Nullable String getLocalizedMessage() {
```

Comment:
Can 6ou add a small unit test for the two important methods?

Reply:
Done